### DISCUSSION TITLE
Passing array variables to blueprints

### DISCUSSION THREAD

**Discussion by CK85** [nxrighthere/UnrealCLR#112] [General]
Hi nxrighthere,

great work! Is there an easy way to set an array via C# in a blueprint e.g.:

			var levelScript = World.GetActor<LevelScript>();
			Vector4[] array = new Vector4[] { new Vector4(1, 1, 1, 1), new Vector4(2, 2, 2, 2) };
			levelScript.SetVector4Array("Vector4Array", array );

Thanks!
Chris

**Comment by nxrighthere** [comment]
This is not supported at the moment.

**Reply by dagillespie** [reply]
would a work around be to pass this through as a string right now and convert it on the BP side?

**Reply by CK85** [reply]
> would a work around be to pass this through as a string right now and convert it on the BP side?

Yes, but sounds inefficient. I will wait till nxrighthere extents the variables parser functions. Thanks anyway for the quick reply!

**Reply by nxrighthere** [reply]
I'll solve it optimally by reconstructing arrays on the native side from managed code similar to how batched operations like [`InstancedStaticMeshComponent.BatchUpdateInstanceTransforms()`](https://github.com/nxrighthere/UnrealCLR/blob/master/API/InstancedStaticMeshComponent-BatchUpdateInstanceTransforms(int_Transform--_bool_bool_bool).md) are done.

**Reply by dagillespie** [reply]
> Yes, but sounds inefficient.

Totally agree. Thanks @nxrighthere for adding this to the backlog

**Reply by dagillespie** [reply]
Have just checked the other way and (as one would expect) arrays from BP -> UnrealCLR isn't supported either yet.
